protocols/imap/src/main/java/org/apache/james/imap/processor/SearchProcessor.java

```diff
@@ -259,245 +251,6 @@ private ModSeq findHighestModSeq(MailboxSession session, MessageManager mailbox,
-            Criterion afterCrit = SearchQuery.headerDateAfter(ImapConstants.RFC822_DATE, date.toDate(), DateResolution.Day);
-            return SearchQuery.or(onCrit, afterCrit);
-        case TYPE_SEQUENCE_SET:
-            return sequence(key.getSequenceNumbers(), session);
-        case TYPE_SINCE:
-            // Include the date which is used as search param. See IMAP-293
-            return SearchQuery.or(SearchQuery.internalDateOn(date.toDate(), DateResolution.Day), SearchQuery.internalDateAfter(date.toDate(), DateResolution.Day));
-        case TYPE_SMALLER:
-            return SearchQuery.sizeLessThan(key.getSize());
-        case TYPE_SUBJECT:
-            return SearchQuery.headerContains(ImapConstants.RFC822_SUBJECT, key.getValue());
-        case TYPE_TEXT:
-            return SearchQuery.mailContains(key.getValue());
-        case TYPE_TO:
-            return SearchQuery.address(AddressType.To, key.getValue());
-        case TYPE_UID:
-            return uids(key.getUidRanges(), session);
-        case TYPE_UNANSWERED:
-            return SearchQuery.flagIsUnSet(Flag.ANSWERED);
-        case TYPE_UNDELETED:
-            return SearchQuery.flagIsUnSet(Flag.DELETED);
-        case TYPE_UNDRAFT:
-            return SearchQuery.flagIsUnSet(Flag.DRAFT);
-        case TYPE_UNFLAGGED:
-            return SearchQuery.flagIsUnSet(Flag.FLAGGED);
-        case TYPE_UNKEYWORD:
-            return SearchQuery.flagIsUnSet(key.getValue());
-        case TYPE_UNSEEN:
-            return SearchQuery.flagIsUnSet(Flag.SEEN);
-        case TYPE_OLDER:
-            Date withinDate = createWithinDate(key);
-            return SearchQuery.or(SearchQuery.internalDateOn(withinDate, DateResolution.Second), SearchQuery.internalDateBefore(withinDate, DateResolution.Second));
-        case TYPE_YOUNGER:
-            Date withinDate2 = createWithinDate(key);
-            return SearchQuery.or(SearchQuery.internalDateOn(withinDate2, DateResolution.Second), SearchQuery.internalDateAfter(withinDate2, DateResolution.Second));
-        case TYPE_MODSEQ: 
-            session.setAttribute(SEARCH_MODSEQ, true);
-            long modSeq = key.getModSeq();
-            return SearchQuery.or(SearchQuery.modSeqEquals(modSeq), SearchQuery.modSeqGreaterThan(modSeq));
-        default:
-            LOGGER.warn("Ignoring unknown search key {}", type);
-            return SearchQuery.all();
-        }
-    }
-    
-    private Date createWithinDate(SearchKey key) {
-        long seconds = key.getSeconds();
-        long res = System.currentTimeMillis() - seconds * 1000;
-        return new Date(res);
-    }
-
-    /**
-     * Create a {@link Criterion} for the given sequence-sets. 
-     * This include special handling which is needed for SEARCH to not return a BAD response on a invalid message-set. 
-     * See IMAP-292 for more details.
-     */
-    private Criterion sequence(IdRange[] sequenceNumbers, ImapSession session) throws MessageRangeException {
```

Comment:
No tests for the methods you move to SearchKey.scala?

Reply:
I assumed this code is well tested with various layers of tests. Do you think I should not?

Reply:
Well, for me the MPT test suite covers this well enough that's for sure ;-)